### PR TITLE
Content update & fix of Notification in Service Worker

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

move the note in `ServiceWorkerGlobalScope.notificationclose` event to `Notification()` constructor

and update content indicating relative event on `Notification` in `ServiceWorkerGlobalScope.notificationclick`

also update example description by replacing event handler property usage to event

### CODE REVIEW

File: files/en-us/web/api/serviceworkerregistration/shownotification/index.md [hunk 1]
```diff
@@ -38,14 +38,14 @@ showNotification(title, options)
           - : A string identifying a user action to be displayed on the notification.
         - `title`
           - : A string containing action text to be shown to the user.
-        - `icon`
+        - `icon` {{optional_inline}}
```
Comment (hamishwillee) [inline files/en-us/web/api/serviceworkerregistration/shownotification/index.md:41]: Flyby. If the parent is optional, aren't all of the sub items (i.e why not action and title too)? Or are you saying that IFF `actions` is specified, then the `action` and `title` must be specified, but `icon` is optional?

Comment (wbamberg) [reply files/en-us/web/api/serviceworkerregistration/shownotification/index.md:41]: > IFF actions is specified, then the action and title must be specified, but icon is optional?

Basically, yes:
- `options` is optional (https://notifications.spec.whatwg.org/#service-worker-api)
-  `options` has an optional attribute `actions` (https://notifications.spec.whatwg.org/#dictdef-notificationoptions)
- `actions` is an array of `NotificationAction` objects, which have mandatory `action` and `title` properties, and an optional `icon` property (https://notifications.spec.whatwg.org/#dictdef-notificationaction)

File: files/en-us/web/api/notification/notification/index.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ browser-compat: api.Notification.Notification
 The **`Notification()`** constructor creates a new
 {{domxref("Notification")}} object instance, which represents a user notification.
 
+> **Note:** Trying to create a notification inside the {{domxref("ServiceWorkerGlobalScope")}} using the `Notification()` constructor will throw a `TypeError`, you should use {{domxref("ServiceWorkerRegistration.showNotification()")}} instead.
```
Comment (wbamberg) [inline files/en-us/web/api/notification/notification/index.md:14]:
```suggestion
> **Note:** Trying to create a notification inside the {{domxref("ServiceWorkerGlobalScope")}} using the `Notification()` constructor will throw a `TypeError`. Use {{domxref("ServiceWorkerRegistration.showNotification()")}} instead.
```

[comma splice](https://en.wikipedia.org/wiki/Comma_splice)

File: files/en-us/web/api/serviceworkerregistration/shownotification/index.md (hunk 1 above)
Comment (wbamberg) [reply files/en-us/web/api/serviceworkerregistration/shownotification/index.md:41]: Although weirdly it seems as if, if you call this constructor, you must not pass any actions, or you get a TypeError exception? https://notifications.spec.whatwg.org/#create-a-notification

File: files/en-us/web/api/notifications_api/using_the_notifications_api/index.md [hunk 3]
```diff
@@ -2,7 +2,11 @@
 title: Using the Notifications API
 slug: Web/API/Notifications_API/Using_the_Notifications_API
 page-type: guide
-browser-compat: api.Notification
+browser-compat:
+  - api.Notification
+  - api.ServiceWorkerRegistration.showNotification
+  - api.ServiceWorkerRegistration.getNotifications
```
Comment (wbamberg) [inline files/en-us/web/api/notifications_api/using_the_notifications_api/index.md:8]: Guide pages don't have Browser compatibility sections. The section and the front matter key should be removed.

File: files/en-us/web/api/notifications_api/using_the_notifications_api/index.md [hunk 4]
```diff
@@ -2,7 +2,11 @@
...
+spec-urls: https://notifications.spec.whatwg.org/
```
Comment (wbamberg) [inline files/en-us/web/api/notifications_api/using_the_notifications_api/index.md:9]: Same for "Specifications".

File: files/en-us/web/api/serviceworkerglobalscope/notificationclick_event/index.md [hunk 5]
```diff
@@ -24,13 +29,13 @@ onnotificationclick = (event) => {};
...
 
-A {{domxref("NotificationEvent")}}. Inherits from {{domxref("Event")}}.
+A {{domxref("NotificationEvent")}}. Inherits from {{domxref("ExtendableEvent")}} and {{domxref("Event")}}.
 
 {{InheritanceDiagram("NotificationEvent")}}
 
 ## Event properties
 
-_Inherits properties from its ancestor, {{domxref("Event")}}_.
+_Inherits properties from its ancestor, {{domxref("ExtendableEvent")}} and {{domxref("Event")}}_.
```
Comment (wbamberg) [inline files/en-us/web/api/serviceworkerglobalscope/notificationclick_event/index.md:38]:
```suggestion
_Inherits properties from its ancestors, {{domxref("ExtendableEvent")}} and {{domxref("Event")}}_.
```

**Review by wbamberg — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/serviceworkerregistration/shownotification/index.md (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/web/api/serviceworkerregistration/shownotification/index.md:41]: Thanks. 

> Although weirdly it seems as if, if you call this constructor, you must not pass any actions, or you get a TypeError exception? https://notifications.spec.whatwg.org/#create-a-notification

It says "If a serviceWorkerRegistration was not provided and options["[actions](https://notifications.spec.whatwg.org/#dom-notificationoptions-actions)"] is not [empty](https://infra.spec.whatwg.org/#list-empty), then [throw](https://webidl.spec.whatwg.org/#dfn-throw) a [TypeError](https://webidl.spec.whatwg.org/#exceptiondef-typeerror)."

So you you can only have actions if the serviceworker registration is provided.

Looking through this there are lots of missing cases that can throw a typerror

Comment (wbamberg) [reply files/en-us/web/api/serviceworkerregistration/shownotification/index.md:41]: > So you you can only have actions if the serviceworker registration is provided.

Right, but that operation I linked, https://notifications.spec.whatwg.org/#create-a-notification , can I think be reached from two paths:
- calling the `Notification()` constructor (https://notifications.spec.whatwg.org/#ref-for-create-a-notification)
- calling `ServiceWorkerRegistration.showNotification()` (https://notifications.spec.whatwg.org/#ref-for-create-a-notification%E2%91%A0)

...and it's only in the second of these cases that the registration gets passed. So if you call the constructor, you must not provide actions.

Comment (skyclouds2001) [reply files/en-us/web/api/serviceworkerregistration/shownotification/index.md:41]: > Right, but that operation I linked, https://notifications.spec.whatwg.org/#create-a-notification , can I think be reached from two paths:
> 
> * calling the `Notification()` constructor (https://notifications.spec.whatwg.org/#ref-for-create-a-notification)
> * calling `ServiceWorkerRegistration.showNotification()` (https://notifications.spec.whatwg.org/#ref-for-create-a-notification%E2%91%A0)
> 
> ...and it's only in the second of these cases that the registration gets passed. So if you call the constructor, you must not provide actions.

yes, `actions` parameter can not use with `Notification()` constructor

**Review by wbamberg — APPROVED** [review]
👍 thank you!

**Pull request merged**